Add more intuitive explanation and blackbox (#23)
privacy-scaling-explorations/zk-contribution-program: MODULE_3.md

## Pairing-Based Cryptography (PBC)

Many of the following topics will depend upon what is called pairing-based cryptography. 

> You can imagine pairing is the multiplication of elliptic curve. The original elliptic curve is additive homomorphic but not multiplicative homomorphic. Pairing is a way to mimic it.

Reviewer: ```suggestion
You can imagine pairing as the multiplication of elliptic curves. The original elliptic curve operation is homomorphically additive but not homomorphically multiplicative. Pairing is a way to mimic this "multiplication".
```

> This is largely used in zk, especially during verification stage. 

Reviewer: ```suggestion
This is largely used in zk, especially during the verification stage. 
```

![Pairing](./assets/elliptic-curve-pairings.jpeg)
[Source](https://www.inevitableeth.com/home/concepts/elliptic-curve-pairings)

> If you like to know more about it, you can check out the following resources:

Reviewer: ```suggestion
If you want to know more about it, check out the following resources:
```

- [Pairings in Cryptography by Dan Boneh [1:29:48]](https://www.youtube.com/watch?v=8WDOpzxpnTE)
    - **Note:** This video is very long and technical, but can be a helpful reference.

> ## BLS Signatures 

Reviewer: ```suggestion
## BLS Signatures
```

BLS (Boneh-Lynn-Shacham) signatures are a type of cryptographic signature scheme that allows for efficient aggregation of individual signatures into a single signature. It also makes use of pairing-based cryptography.

> The intuition of BLS is that it is an aggregation signature. The image below shows the public key and signature are aggregated. The advantage is all the signature can be verified at once. That's why it is used in consensus protocol.

Reviewer: ```suggestion
The intuition of BLS is that it is an aggregation signature. The image below shows how the public key and signature are aggregated. The advantage is that all signatures can be verified at once. That's why it is commonly used in consensus protocols.
```

The concept of a trusted setup is an important part of the KZG Polynomial commitment scheme, and indeed part of the wider culture of Ethereum. Here are a few resources to learn more about trusted setups.

> The intuition of trusted setup is that it is a process to generate a bunch of point on the elliptic curve, the prover will input these points to the polynomial commitment. Therefore the output is a point on the curve as well. Especially the generation of these points requires many people join and discard the secret value. That's why it is called trusted.

Reviewer: ```suggestion
The intuition of a trusted setup is that it is a process to generate a bunch of points on the elliptic curve, and the prover will input these points to the polynomial commitment. Therefore the output is a point on the curve as well. Specifically, the generation of these points require many people join and discard the secret value (which they used to create the point). It should not be possible to generate the points without this. That's why it is called a trusted setup.
```
